opts/envdir.go

```diff
@@ -0,0 +1,56 @@
+package opts
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+/*
+Read in a directory with files acting as environment variables.
+*/
+func ParseEnvDir(directory string) ([]string, error) {
+	dir, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return []string{}, err
+	}
+
+	lines := []string{}
+	for _, info := range dir {
+		variable := info.Name()
+		// ignore directories and hidden files
+		if !info.IsDir() && variable[0] != '.' {
+			if strings.ContainsAny(variable, whiteSpaces) {
+				return []string{}, ErrBadEnvVariable{fmt.Sprintf("variable '%s' has white spaces", variable)}
+			}
+
+			data, err := getFirstLine(filepath.Join(directory, variable))
+			if err != nil {
+				return []string{}, err
+			}
+
+			lines = append(lines, fmt.Sprintf(
+				"%s=%s",
+				variable,
+				// null bytes are to be converted to newlines according to envdir
+				bytes.TrimSpace(bytes.Replace(data, []byte{0}, []byte{'\n'}, -1)),
```

Comment:
per the doc you referenced, \0 is replace with \n but that seems kind of weird and not very user-friendly.
Is that doc a standard or just one person's view of how this should be done?
It just seems to me that rather than converting \0 to \n and ONLY using the first line of the file, why not read in the entire file 'as is' and let the user put line-breaks in there instead of \0's.